Move `position` from `NonInheritedFlags` to `StyleBoxData`

nt1m opened the pull request:
#### 4ef4a110000dcf0223e2b3631f30ff6d567ecc16
<pre>
Move `position` from `NonInheritedFlags` to `StyleBoxData`
<a href="https://bugs.webkit.org/show_bug.cgi?id=269410">https://bugs.webkit.org/show_bug.cgi?id=269410</a>
<a href="https://rdar.apple.com/122976910">rdar://122976910</a>

Reviewed by Aditya Keerthi and Darin Adler.

Free-up 3 bits from `NonInheritedFlags` for `::target-text` and `text-decoration-line: spelling-error/grammar-error`.

* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::RenderStyle):
(WebCore::RenderStyle::NonInheritedFlags::copyNonInheritedFrom):
(WebCore::RenderStyle::hashForTextAutosizing const):
(WebCore::RenderStyle::equalForTextAutosizing const):
(WebCore::RenderStyle::changeRequiresLayout const):
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::position const): Deleted.
(WebCore::RenderStyle::setPosition): Deleted.
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::position const):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setPosition):
* Source/WebCore/rendering/style/StyleBoxData.cpp:
(WebCore::StyleBoxData::StyleBoxData):
(WebCore::StyleBoxData::operator== const):
* Source/WebCore/rendering/style/StyleBoxData.h:
(WebCore::StyleBoxData::position const):

Canonical link: <a href="https://commits.webkit.org/274683@main">https://commits.webkit.org/274683@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3d18be2fa061eb4c1aa5837e23f13049ff9507de

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39764 "Passed style check") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18743 "Hash 3d18be2f for PR 24462 does not build (failure)") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42120 "Hash 3d18be2f for PR 24462 does not build (failure)") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/42299 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/35666 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/42071 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/21650 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16072 "Hash 3d18be2f for PR 24462 does not build (failure)") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/42299 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40338 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/21650 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/14/builds/42120 "Hash 3d18be2f for PR 24462 does not build (failure)") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/42299 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/21650 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/14/builds/42120 "Hash 3d18be2f for PR 24462 does not build (failure)") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/43576 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/21650 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/14/builds/42120 "Hash 3d18be2f for PR 24462 does not build (failure)") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/43576 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/14579 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/51/builds/16072 "Hash 3d18be2f for PR 24462 does not build (failure)") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/43576 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/16198 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/14/builds/42120 "Hash 3d18be2f for PR 24462 does not build (failure)") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/16246 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5226 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/15842 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/rendering/style/RenderStyle.h:
```diff
@@ -1104,7 +1104,7 @@ class RenderStyle {
     }
     void setEffectiveDisplay(DisplayType v) { m_nonInheritedFlags.effectiveDisplay = static_cast<unsigned>(v); }
-    void setPosition(PositionType v) { m_nonInheritedFlags.position = static_cast<unsigned>(v); }
+    void setPosition(PositionType v);
```
This should be marked `inline` so we get an error if we forget to include `RenderStyleSetters.h`.

I suggest removing the variable name `v`.

darinadler commented on Source/WebCore/rendering/style/RenderStyle.h:
```diff
@@ -426,7 +426,7 @@ class RenderStyle {
     inline bool hasStaticBlockPosition(bool horizontal) const;
 
-    PositionType position() const { return static_cast<PositionType>(m_nonInheritedFlags.position); }
+    PositionType position() const;
```
This should be marked `inline` so we get an error if we forget to include `RenderStyleInlines.h`.

nt1m committed:
Move `position` from `NonInheritedFlags` to `StyleBoxData`
https://bugs.webkit.org/show_bug.cgi?id=269410
rdar://122976910

Reviewed by Aditya Keerthi and Darin Adler.

Free-up 3 bits from `NonInheritedFlags` for `::target-text` and `text-decoration-line: spelling-error/grammar-error`.

* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::RenderStyle):
(WebCore::RenderStyle::NonInheritedFlags::copyNonInheritedFrom):
(WebCore::RenderStyle::hashForTextAutosizing const):
(WebCore::RenderStyle::equalForTextAutosizing const):
(WebCore::RenderStyle::changeRequiresLayout const):
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::position const): Deleted.
(WebCore::RenderStyle::setPosition): Deleted.
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::position const):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setPosition):
* Source/WebCore/rendering/style/StyleBoxData.cpp:
(WebCore::StyleBoxData::StyleBoxData):
(WebCore::StyleBoxData::operator== const):
* Source/WebCore/rendering/style/StyleBoxData.h:
(WebCore::StyleBoxData::position const):

Canonical link: https://commits.webkit.org/274683@main
